substratevm/src/com.oracle.objectfile/src/com/oracle/objectfile/ObjectFile.java

```diff
@@ -528,15 +528,19 @@ public Section newUserDefinedSection(String name, ElementImpl impl) {
         return result;
     }
 
-    public Section newDebugSection(String name, ElementImpl impl) {
-        final Segment segment = getOrCreateSegment(null, name, false, false);
+    public Section newDebugSection(String segmentName, String name, ElementImpl impl) {
+        final Segment segment = getOrCreateSegment(segmentName, name, false, false);
```

Comment:
Please re-order this new method after the redefined version of original so the changes shown in the diff are clearer.